libyul/optimiser/Suite.cpp

```diff
@@ -267,44 +267,29 @@ map<char, string> const& OptimiserSuite::stepAbbreviationToNameMap()
 	return lookupTable;
 }
 
+
+
 void OptimiserSuite::validateSequence(string const& _stepAbbreviations)
 {
+	stack<char> s;
+   	char ch;
 	bool insideLoop = false;
-	for (char abbreviation: _stepAbbreviations)
-		switch (abbreviation)
-		{
-		case ' ':
-		case '\n':
-			break;
-		case '[':
-			assertThrow(!insideLoop, OptimizerException, "Nested brackets are not supported");
-			insideLoop = true;
-			break;
-		case ']':
-			assertThrow(insideLoop, OptimizerException, "Unbalanced brackets");
-			insideLoop = false;
-			break;
-		default:
-		{
-			yulAssert(
-				string(NonStepAbbreviations).find(abbreviation) == string::npos,
-				"Unhandled syntactic element in the abbreviation sequence"
-			);
-			assertThrow(
-				stepAbbreviationToNameMap().find(abbreviation) != stepAbbreviationToNameMap().end(),
-				OptimizerException,
-				"'"s + abbreviation + "' is not a valid step abbreviation"
-			);
-			optional<string> invalid = allSteps().at(stepAbbreviationToNameMap().at(abbreviation))->invalidInCurrentEnvironment();
-			assertThrow(
-				!invalid.has_value(),
-				OptimizerException,
-				"'"s + abbreviation + "' is invalid in the current environment: " + *invalid
-			);
-
-		}
-		}
-	assertThrow(!insideLoop, OptimizerException, "Unbalanced brackets");
+	int stepAbb_length = _stepAbbreviations.size();
```

Comment:
Our naming convention is camel case:
```suggestion
	int stepAbbLength = _stepAbbreviations.size();
```

Also, a name using full words would be better: `stepAbbreviationsLength` (or even better just `sequenceLength`).